### PR TITLE
Move to "salted" interface and remove deterministic signature support

### PR DESCRIPTION
**Pull request by FredericJacobs**
Some edits reflecting new findings regarding the proof of security regarding the blindness property.

### CODE REVIEW

File: draft-irtf-cfrg-rsa-blind-signatures.md
Comment (FredericJacobs) [inline draft-irtf-cfrg-rsa-blind-signatures.md:68]: The paper should be live ahead of the next IETF meeting draft upload deadline

File: draft-irtf-cfrg-rsa-blind-signatures.md
Comment (FredericJacobs) [inline draft-irtf-cfrg-rsa-blind-signatures.md:572]: A TODO is also to add test vectors with the additional hashing step.

File: draft-irtf-cfrg-rsa-blind-signatures.md
Comment (jedisct1) [inline draft-irtf-cfrg-rsa-blind-signatures.md:244]: Maybe `msg_salt` would be better here, since `salt` is not an adjective?

**Review by chris-wood — CHANGES_REQUESTED** [review]
I'm not overly thrilled about this change as-is. In particular, it changes the default API for all applications, even for those which may not need it. The proposed text that suggests applications can omit this message salting step with high-entropy input is kind of confusing. 

Instead, I would split out `blind`  and `finalize` with salting into a _separate_ API, e.g., `salted_blind` and `salted_finalize`, or something, and then give applications guidance as to which one should be used. That guidance might look something like the following:

~~~
The choice of blinding mechanism has security implications. {{XXX}} analyzes the security properties of `blind` in the presence of a malicious signer with low-entropy inputs. The results from this analysis can be summarized as follows:

- Blinding with message salting is safe for all application use cases.
- Blinding without message salting is possibly unsafe, unless one of the following conditions are met:
	1. The client has proof of the signer's public key being a product of two primes, or
	2. The client input has high entropy.
~~~

File: draft-irtf-cfrg-rsa-blind-signatures.md
Comment (jedisct1) [reply draft-irtf-cfrg-rsa-blind-signatures.md:572]: I have an update to the Python code that generates the test vectors, that I can commit right after this PR got merged.

File: draft-irtf-cfrg-rsa-blind-signatures.md
Comment (chris-wood) [reply draft-irtf-cfrg-rsa-blind-signatures.md:68]: Huzzah!

**Comment by jedisct1** [comment]
> Wait, what? Blinding with a salt length of zero will produce a deterministic signature, right?
> […](#)

That PR removes support for empty salts.

**Comment by chris-wood** [comment]
> That PR removes support for empty salts.

Right, as I'm trying to show, I think removing empty salts is separate from whether we also provide a "salted message" interface.

**Comment by jedisct1** [comment]
Alternatively, maybe we can define a `Prehash(msg_salt, msg)` function, that applications will apply to the message prior to calling `Finalize()` or `Verify()` when the input doesn't have enough entropy.

**Comment by chris-wood** [comment]
@FredericJacobs I updated this PR based on #107 -- please have a look!

**Review by chris-wood — APPROVED** [review]
Approving since I think this is the preferred path forward, but let's please not merge until there's been RG discussion.

**Comment by FredericJacobs** [comment]
We will be closing this and replace it with #106

**Pull request closed without merging**
